tests/win_named_pipe.rs

```diff
@@ -0,0 +1,224 @@
+#![cfg(windows)]
+
+use std::fs::OpenOptions;
+use std::io;
+use std::os::windows::fs::*;
+use std::os::windows::io::*;
+use std::task::{Context, Poll};
+use std::time::Duration;
+
+// use mio::{Events, Poll, PollOpt, Ready, Token};
+use mio::windows::NamedPipe;
+use rand::Rng;
+use winapi::um::winbase::*;
+
+use futures_test::task::new_count_waker;
+
+macro_rules! t {
+    ($e:expr) => {
+        match $e {
+            Ok(e) => e,
+            Err(e) => panic!("{} failed with {}", stringify!($e), e),
+        }
+    };
+}
+
+fn server(registry: &mio::Registry) -> (NamedPipe, String) {
+    let num: u64 = rand::thread_rng().gen();
+    let name = format!(r"\\.\pipe\my-pipe-{}", num);
```

Comment:
Nit: we could just use the test name here, then we don't need to add `rand` as dev-dependency.

Reply:
Unless the tests are run in multiple threads :P

Reply:
(multiple processes)

Reply:
True, but we already use something like this in `temp_file` (in `tests/util/mod.rs::240`).